bug(agent): idle loop 탐지 — 5 consecutive identical tool calls (6회/일)

## 증상

Agent가 동일한 tool call을 5회 연속 반복하여 idle 탐지에 걸림.

## 로그 증거 (system_log_2026-03-30.jsonl)

```
[ERROR] unified turn failed: Agent run failed: Idle detected: 5 consecutive identical tool call turns  x6
```

## 분석

- 3/30에 6회 발생, 다른 날짜에는 미발생
- Agent가 tool call 결과를 처리하지 못하고 같은 호출을 반복하는 패턴
- 가능한 원인:
  1. Tool 응답이 에러를 반환하지만 agent가 다른 전략으로 전환하지 못함
  2. Tool 응답 파싱 실패로 동일 요청 재시도
  3. Context에 이전 결과가 반영되지 않는 상태 관리 버그

## 예상 영향

- Agent turn 낭비 (토큰 비용)
- Keeper가 이 상태에 빠지면 진행 불가

## 제안

- Idle 탐지 시 어떤 tool이 반복되었는지 로깅 (현재 tool name 미기록)
- 3회 반복 시점에 tool 응답 내용을 agent context에 명시적으로 주입하는 escape hatch
- idle 후 recovery 전략 (다른 tool 시도 또는 사용자 알림)

---
출처: system_log_2026-03-30.jsonl, 자동 로그 모니터링

--- a/lib/keeper/keeper_hooks_oas.ml
+++ b/lib/keeper/keeper_hooks_oas.ml
@@ -290,9 +290,17 @@ let make_hooks
 
     on_idle = Some (fun event ->
       match event with
-      | Agent_sdk.Hooks.OnIdle { consecutive_idle_turns; _ } ->
-        Log.Keeper.debug "keeper:%s idle_turns=%d"
-          (!meta_ref).name consecutive_idle_turns;
+      | Agent_sdk.Hooks.OnIdle { consecutive_idle_turns; tool_names } ->
+        let tools_str = match tool_names with
+          | [] -> "<none>"
+          | names -> String.concat ", " names
+        in
+        if consecutive_idle_turns >= 3 then
+          Log.Keeper.warn "keeper:%s idle_turns=%d repeated_tools=[%s]"
+            (!meta_ref).name consecutive_idle_turns tools_str
+        else
+          Log.Keeper.debug "keeper:%s idle_turns=%d tools=[%s]"
+            (!meta_ref).name consecutive_idle_turns tools_str;
         Agent_sdk.Hooks.Continue
       | _ -> Agent_sdk.Hooks.Continue);
   }

